includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Class WC_Payments_Payment_Method_Messaging_Element
+ *
+ * @package WooCommerce\Payments
+ */
+
+declare( strict_types=1 );
+
+use WCPay\Constants\Payment_Method;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+use WCPay\Payment_Methods\UPE_Payment_Gateway;
+use WCPay\Payment_Methods\UPE_Split_Payment_Gateway;
+/**
+ * WC_Payments_Payment_Method_Messaging_Element class.
+ */
+class WC_Payments_Payment_Method_Messaging_Element {
+	/**
+	 * WC_Payments_Account instance to get information about the account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+	/**
+	 * WC_Payments_Gateway instance to get information about the enabled payment methods.
+	 *
+	 * @var UPE_Payment_Gateway|UPE_Split_Payment_Gateway
+	 */
+	private $gateway;
+
+	/**
+	 * WC_Payments_Payment_Method_Messaging_Element constructor
+	 *
+	 * @param  WC_Payments_Account                           $account Account instance.
+	 * @param  UPE_Payment_Gateway|UPE_Split_Payment_Gateway $gateway Gateway instance.
+	 * @return void
+	 */
+	public function __construct( WC_Payments_Account $account, $gateway ) {
+		$this->account = $account;
+		$this->gateway = $gateway;
+	}
+
+	/**
+	 * Initializes the payment method messaging element.
+	 *
+	 * @return string The HTML markup for the payment method message container.
+	 */
+	public function init(): string {
+		global $product;
+		$price         = $product->get_price();
+		$currency_code = get_woocommerce_currency();
+
+		if ( WC()->customer ) {
+			$billing_country = WC()->customer->get_billing_country();
+		}
+		if ( ! $billing_country ) {
+			$billing_country = WC()->countries->get_base_country(); // Fallback to merchant's country.
+		}
+
+		$enabled_upe_payment_methods = $this->gateway->get_payment_method_ids_enabled_at_checkout();
+		// Filter non BNPL out of the list of payment methods.
+		$bnpl_payment_methods = array_filter(
```

Comment:
It sounds like what 'array_intersec' does.

Reply:
Addressed here [55c6ff2](https://github.com/Automattic/woocommerce-payments/pull/6522/commits/55c6ff25ddbeb78cc19c452d3aacf33eb1fbdefd) thanks!